core/platform/src/command_return.rs

```diff
@@ -0,0 +1,163 @@
+use crate::{return_type, ErrorCode, ReturnType};
+
+/// The response type from `command`. Can represent a successful value or a
+/// failure.
+#[derive(Clone, Copy)]
+pub struct CommandReturn {
+    pub(crate) return_type: ReturnType,
```

Comment:
This more a question than feedback: I've come to call the different return types "variants", akin to the Rust terminology around enums. I see that the TRD uses the notion of _types_ both when referring to the different system calls (which I agree technically have different types, e.g. an `AllowReturn` or `CommandReturn`), and when referring to the different _variants_ indicated in `r0`. What do you think? I think it's beneficial to have a clear distinction here, and going in terms of the Rust terminology seems to be a good option. I'd update the TRD if this naming scheme makes sense.

Reply:
I agree with that! I struggled with naming `ReturnType` for this reason, and I think `ReturnVariant` would be a better name.

If we make that terminology change, I'll take care of renaming `ReturnType`.

Reply:
I like this more precise terminology and am in favor.

Reply:
Alright, I'll make a PR to update the TRD accordingly.

Reply:
You can just make the changes. There will be lots of editing passes between now and then.

Reply:
FYI: I have pushed the updated TRD to refer to "system call return variants" (when talking about e.g. Success, FailureU32, etc.). So this should now be synced with that document.